testbed/HIL_Firmware/common/dac.c

```diff
@@ -0,0 +1,140 @@
+#include <stdio.h>
+#include "dac.h"
+#include "freertos/FreeRTOS.h"
+#include "esp_err.h"
+#include "driver/twai.h"
+#include "driver/spi_master.h"
+#include "driver/dac_oneshot.h"
+
+spi_device_handle_t throttle_A;
+spi_device_handle_t throttle_B;
+spi_device_handle_t brake_pos;
+spi_device_handle_t steer_raw;
+
+//retain 4 least significant bits
+static uint8_t Vout_mask = 0X0F;
+
+twai_message_t message_status = {
+    .identifier = MESSAGE_STATUS,
+    .extd = 1,
+    .data_length_code = 1,
+};
+
+//convert VREF from mV to V
+const double STEPV12 = (VREF/1000)/MAXSTEPS12;
+const double STEPV8 = (VREF/1000)/MAXSTEPS8;
+
+static bool channel1=false;
+
+static dac_oneshot_handle_t chan0_handle;
+static dac_oneshot_config_t chan0_cfg = {
+    .chan_id = DAC_CHAN_0,
+};
+
+int setDacVoltage(float voltage)
+{
+    esp_err_t fault = 0;
```

Comment:
Ideally the faults should be an enum but I do understand if you didn't create the esp_err_t